src/v/datalake/tests/datalake_parquet_tests.cc

```diff
@@ -532,3 +535,373 @@ TEST(DatalakeParquetSchema, NestedStruct) {
     ASSERT_TRUE(
       primitive_schema_matches(parquet_schema.children[0], primitive_types()));
 }
+
+/**
+ * Helper to determine parquet physical type value for an iceberg value
+ */
+template<typename ParquetT>
+struct type_wrapper {
+    using type = ParquetT;
+};
+
+template<typename T>
+static constexpr auto parquet_type(const T&) {
+    if constexpr (std::is_same_v<T, iceberg::boolean_value>) {
+        return type_wrapper<serde::parquet::boolean_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::int_value>) {
+        return type_wrapper<serde::parquet::int32_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::long_value>) {
+        return type_wrapper<serde::parquet::int64_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::float_value>) {
+        return type_wrapper<serde::parquet::float32_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::double_value>) {
+        return type_wrapper<serde::parquet::float64_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::decimal_value>) {
+        return type_wrapper<serde::parquet::fixed_byte_array_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::date_value>) {
+        return type_wrapper<serde::parquet::int32_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::time_value>) {
+        return type_wrapper<serde::parquet::int64_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::timestamp_value>) {
+        return type_wrapper<serde::parquet::int64_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::timestamptz_value>) {
+        return type_wrapper<serde::parquet::int64_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::string_value>) {
+        return type_wrapper<serde::parquet::byte_array_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::uuid_value>) {
+        return type_wrapper<serde::parquet::fixed_byte_array_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::fixed_value>) {
+        return type_wrapper<serde::parquet::fixed_byte_array_value>{};
+    }
+    if constexpr (std::is_same_v<T, iceberg::binary_value>) {
+        return type_wrapper<serde::parquet::byte_array_value>{};
+    }
+    __builtin_unreachable();
```

Comment:
@mmaslankaprv this should use dependent_false so that it is a compile error rather than a runtime error.